### PR TITLE
COMPAT: Date format for object copy and copy part

### LINKED ISSUE #372: [ObjectCopy][XMLResponse] bad Date format in LastModified field
Testing hadoop against the S3 server, the java code fails when attempting to parse a Date. It looks like it's attemtping to parse it as Iso8601 when it's not formatted this way.

Here is the S3 Logs (trace level) from s3server:
```
{"name":"S3","bucketName":"mybucket","objectKey":"derby93","accessKey":"accessKey1","xml":"<?xml version=\"1.0\" encoding=\"UTF-8\"?><CopyObjectResult><LastModified>Wed, 09 Nov 2016 16:16:12 GMT</LastModified><ETag>&quot;d88d58e233cfba321424c8452ba2a314&quot;</ETag></CopyObjectResult>","time":1478708172757,"req_id":"162a10e21335c36b3f42","level":"trace","message":"xml response","hostname":"c314ca580bc8","pid":57}
```

Here is the output from a s3cmd ObjectCopy
```
DEBUG: Response: {'status': 200, 'headers': {'content-length': '234', 'via': '1.1 ubnt1 (squid/3.4.8)', 'x-cache': 'MISS from ubnt1', 'x-amz-id-2': 'NGblUtou6K8lvXGvmQW4tcZEsnJxRm046qLBhMJTR8cpHRrMaLGcFheFWHSrAdO0wK81auZHH4E=', 'x-cache-lookup': 'MISS from ubnt1:0', 'server': 'AmazonS3', 'connection': 'keep-alive', 'x-amz-request-id': '568CD32D9EF08BD1', 'date': 'Wed, 09 Nov 2016 16:24:52 GMT', 'content-type': 'application/xml'}, 'reason': 'OK', 'data': '<?xml version="1.0" encoding="UTF-8"?>\n<CopyObjectResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><LastModified>2016-11-09T16:24:52.000Z</LastModified><ETag>&quot;fac2a668031954664ec5febf2f301c27&quot;</ETag></CopyObjectResult>'}
```

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
Fixes #372

### CODE REVIEW

**Comment by LaurenSpiegel** [comment]
@ironman-machine work it

**Comment by LaurenSpiegel** [comment]
end to end passed

File: tests/functional/s3curl/tests.js [hunk 1]
```diff
@@ -865,55 +865,133 @@ describe('s3curl object ACLs', () => {
...
+                    parseString(rawOutput.stdout, (err, result) => {
+                        if (err) {
+                            assert.ifError(err);
+                        }
+                        uploadId =
+                        result.InitiateMultipartUploadResult.UploadId[0];
+                        // create file to copy
+                        createFile(upload, 100, () => {
+                            // put file to copy
+                            provideRawOutput([
```
Comment (DavidPineauScality) [inline tests/functional/s3curl/tests.js]: you could probably use "putObjects" here

File: tests/functional/s3curl/tests.js [hunk 2]
```diff
@@ -865,55 +865,133 @@ describe('s3curl object ACLs', () => {
...
                     assert.strictEqual(httpCode, '200 OK');
-                    parseString(rawOutput.stdout, (err, result) => {
-                        assert.strictEqual(result.ListPartsResult.UploadId[0],
-                                           uploadId);
-                        assert.strictEqual(result.ListPartsResult.Bucket[0],
-                                           bucket);
-                        assert.strictEqual(result.ListPartsResult.Key[0], key);
-                        assert.strictEqual(result.ListPartsResult.Part,
-                                           undefined);
+                    provideRawOutput([
```
Comment (DavidPineauScality) [inline tests/functional/s3curl/tests.js]: same, you may wanna use `putObjects` here

Comment (LaurenSpiegel) [inline tests/functional/s3curl/tests.js]: Need to send in additional header so can't use the function.

File: tests/functional/s3curl/tests.js [hunk 3]
```diff
@@ -865,55 +865,125 @@ describe('s3curl object ACLs', () => {
...
+            ['--', `${bucketPath}/${key}?uploadId=${uploadId}&partNumber=1`,
+            '-X', 'PUT', '-H',
+            `x-amz-copy-source:${bucket}/copyme`, '-v'],
+            (httpCode, rawOutput) => {
+                assert.strictEqual(httpCode, '200 OK');
+                parseString(rawOutput.stdout, (err, result) => {
+                    const lastModified = result.CopyPartResult
+                        .LastModified[0];
+                    const isoDateString = new Date(lastModified).toISOString();
+                    assert.strictEqual(lastModified, isoDateString);
```
Comment (bennettbuchanan) [inline tests/functional/s3curl/tests.js]: Nice!

File: tests/functional/s3curl/tests.js (hunk 2 above)
Comment (DavidPineauScality) [inline tests/functional/s3curl/tests.js]: Which one ? I do not see it in the list of parameters given to  `provideRawOutput`. Did I miss something ?

Comment (LaurenSpiegel) [inline tests/functional/s3curl/tests.js]: I thought you were referring to the copy one below.  This one can be switched out.

**Comment by LaurenSpiegel** [comment]
@ironman-machine check it

**Comment by DavidPineauScality** [comment]
Thanks, all good now :)

**Comment by LaurenSpiegel** [comment]
@ironman-machine morning

**Comment by LaurenSpiegel** [comment]
end to end passed: http://ci.ironmann.io/gh/scality/Integration/6207

**Pull request merged**